Assets/MixedRealityToolkit-Examples/Demos/Boundary/Scripts/BoundaryVisualizer.cs

```diff
@@ -38,7 +43,7 @@ private void Start()
         /// </summary>
         private void AddQuad()
         {
-            IMixedRealityBoundarySystem boundaryManager = MixedRealityManager.Instance?.GetManager<IMixedRealityBoundarySystem>();
+            boundaryManager = boundaryManager ?? MixedRealityManager.Instance?.GetManager<IMixedRealityBoundarySystem>();
```

Comment:
Instead of using `MixedRealityManager.Instance` let's confirm the MR Manager is active else we bail early.

Reply:
Look at the ways we get the input system and have a property with that checks if the boundary system is cached in a similar way.

Reply:
Here's how I see the input system being accessed...

>+        private IMixedRealityInputSystem inputSystem = null; 
+        protected IMixedRealityInputSystem InputSystem => inputSystem ?? (inputSystem = MixedRealityManager.Instance.GetManager<IMixedRealityInputSystem>()); 

This looks pretty close to what I originally had (minus the ?? check).

>Instead of using MixedRealityManager.Instance? let's confirm the MR Manager is active else we bail early.

Other than checking the instance, how do you suggest checking that the manager is active? do you mean if the monobehavior is enabled?

Reply:
We're not using monobehaviours for our managers, so that might be a bit difficult  😅 
But there is a way to check if the system is enabled.

Reply:
Question: Should SDK/Example scripts be checking the scene for the MixedRealityManager component? Isnt it more efficient to check the instance parameter (is static, would be null if not in the scene).

Reply:
This line should go away if we're following the pattern above.